File: swd-be/src/main/java/com/example/swd/service/OrderService.java
package com.example.swd.service;

import com.example.swd.dto.response.OrderResponse;
import com.example.swd.entity.Order; // Vẫn cần import nếu có method nội bộ dùng, nhưng API public nên dùng OrderResponse
import org.springframework.data.domain.Page;
import org.springframework.data.domain.Pageable;

import java.util.List;

public interface OrderService {
    OrderResponse getOrderById(String orderId);

    List<OrderResponse> getOrdersByUserId(String userId);

    Page<OrderResponse> getAllOrders(Pageable pageable, String statusFilter);

    OrderResponse updateOrderStatus(String orderId, String newStatus);

    OrderResponse cancelOrder(String orderId);
}

File: swd-fe/src/services/orderService.js
import axiosClient from "../api/axiosClient";

// VITE_API_BASE_URL = http://localhost:8080/swd
// ENDPOINT cần thêm '/api/orders' để thành công thức: http://localhost:8080/swd/api/orders
const ENDPOINT = "/api/orders";

export const getAllOrders = (page = 0, size = 10, status = '') => {
    const params = new URLSearchParams();
    params.append('page', page);
    params.append('size', size);
    if (status) {
        params.append('status', status);
    }
    return axiosClient.get(`${ENDPOINT}?${params.toString()}`);
};

export const getOrdersByUserId = (userId) => {
    return axiosClient.get(`${ENDPOINT}/user/${userId}`);
};

export const getOrderById = (orderId) => {
    return axiosClient.get(`${ENDPOINT}/${orderId}`);
};

export const updateOrderStatus = (orderId, newStatus) => {
    return axiosClient.put(`${ENDPOINT}/${orderId}/status`, { status: newStatus });
};

export const cancelOrder = (orderId) => {
    return axiosClient.delete(`${ENDPOINT}/${orderId}`);
};